src/main/java/seedu/address/model/DeckModelManager.java

```diff
@@ -18,28 +18,27 @@
 /**
  * Represents the in-memory model of the deck data.
  */
-public class CardModelManager implements CardModel {
-    private static final Logger logger = LogsCenter.getLogger(CardModelManager.class);
-
-    private final PowerDeck deck;
+public class DeckModelManager implements DeckModel {
+    private static final Logger logger = LogsCenter.getLogger(DeckModelManager.class);
+    private PowerDeck selectedDeck = null;
+    private final ArrayList<PowerDeck> powerDecks = new ArrayList<>();
     private final UserPrefs userPrefs;
-    private final FilteredList<PowerCard> filteredCards;
+    //  private final FilteredList<PowerCard> filteredCards;
```

Comment:
The filteredCardList is basically a FilterList instance of the current list of card in the current chosen deck. So i guess whenever we switch deck we can create a new FilteredList instance of that deck?

Reply:
Yep I agree, we can initialize it whenever we select a new deck, not at the start.